scout/adapter/mongo/case.py

```diff
@@ -897,41 +911,76 @@ def update_manual_tagged_variants(
                         new_var["institute"], case_obj["display_name"], new_var["_id"]
                     )
 
-                    if tag == "manual_rank":
+                    updated_variant = None
+
+                    if action == "manual_rank":
```

Comment:
This could have been expressed more elegantly in c++. 😄 Should have been sth like action.update() that in turn deals with the polymorphic subtypes. This is hardly the only place we do this though.

Reply:
or in java (switch, case)

Reply:
Yes, certainly! Actually it think it can be done in python as well, but requires some arcane syntax, eg like *kwargs for function arguments when the argument lists differ for the different function calls. This is ok.

Reply:
The "problem" is that we (I? 😊) didn't make the tags one main type, with several subtypes. Then the list would have been one or two ifs and one updated_variant = self.update_tag().